airflow/providers/common/sql/operators/sql.py

```diff
@@ -221,6 +220,16 @@ def execute(self, context):
 
         return output
 
+    def on_kill(self) -> None:
+        for query_id in self._hook.query_ids.copy():
+            self.log.info("Stopping query: %s", self._hook.query_ids)
+            try:
+                self._hook.kill(query_id)
+            except NotImplementedError:
+                self.log.info("Method '.kill()' is not implemented for ", self._hook.__class__.__name__)
+            except Exception as e:
+                self.log.info("The query '%s' can not be killed due to %s", self._hook.query_ids, str(e))
```

Comment:
should be query_id:

```suggestion
                self.log.info("The query '%s' can not be killed due to %s", query_id, str(e))
```
but also, i think you don't need to wrap with str:

```suggestion
                self.log.info("The query '%s' can not be killed due to %s", query_id, e)
```

but also, why not just log the exception?

```suggestion
                self.log.exception("The query '%s' could not be killed.", query_id)
```

It would include the error and traceback.  That seems reasonable no?